### PR TITLE
add NetworkClient tests

### PR DESCRIPTION
**Pull request by uweenukr**

### CODE REVIEW

**Comment by paulpach** [comment]
Please fix code smells

File: Assets/Mirror/Runtime/NetworkClient.cs [hunk 1]
```diff
@@ -349,18 +349,20 @@ public bool RemovePlayer()
         {
             if (LogFilter.Debug) Debug.Log("ClientScene.RemovePlayer() called with connection [" + Connection + "]");
 
-            if (Connection.Identity != null)
-            {
-                Connection.Send(new RemovePlayerMessage());
+            if (Connection == null)
+                return false;
```
Comment (paulpach) [inline Assets/Mirror/Runtime/NetworkClient.cs:353]: Should we throw exception instead?

Comment (uweenukr) [reply Assets/Mirror/Runtime/NetworkClient.cs]: sounds good to me. added an exception and updated the test to check for it.

**Comment by uweenukr** [comment]
5 code smells. fix 4 and leave 1 i had a question about. there are 17?!?? lol

**Pull request merged**